src/lib/drivers/driver.js

```diff
@@ -86,6 +86,28 @@ class DriverBase {
     return Promise.reject(new Error('Not implemented'));
   }
 
+  evaluateAsync(asyncExpression) {
+    return new Promise((resolve, reject) => {
+      // Inject the call to capture inspection.
+      const expression = `window.__inspect = inspect;${asyncExpression}`;
+
+      this.on('Runtime.inspectRequested', value => {
+        // Tidy up the injected call.
+        this.sendCommand('Runtime.evaluate', {
+          expression: 'delete window.__inspect'
+        }).then(_ => resolve(value));
```

Comment:
it might be worth considering adding something like an ID, injected with the `__inspect` reference, and checking if it matches on `inspectRequested` since anything could call `inspect`.

Not sure how realistic this is, since we don't necessarily need to support being reentrant, since we explicitly run our gatherers in series, and other devtools instance can't be connected to the page at the same time, but it's an easy thing to do